bootstrap/src/routes.py

```diff
@@ -0,0 +1,123 @@
+
+logger = logging.getLogger('bootstrap')
+
+# Has no effect, just to help the IDE a bit :)
+app.kin_client: kin.KinClient
+app.kin_account: kin.KinAccount
+
+
+@app.route('/balance/<address>', methods=['GET'])
+async def get_balance(request, address: str):
+    balance_request = requets_models.BalanceRequest(address=address)
+    try:
+        balance = await app.kin_client.get_account_balance(balance_request.address)
+    except KinErrors.AccountNotFoundError:
+        raise errors.AccountNotFoundError(balance_request.address)
+
+    return json_response(responses_models.BalanceResponse(balance).to_response_dict(), 200)
+
+
+@app.route('/payment/<tx_hash>', methods=['GET'])
+async def get_tx_info(request, tx_hash: str):
+    tx_info_request = requets_models.TransactionInfoRequest(tx_hash=tx_hash)
+    try:
+        tx = await app.kin_client.get_transaction_data(tx_info_request.tx_hash)
+    except KinErrors.ResourceNotFoundError:
+        raise errors.TransactionNotFoundError(tx_info_request.tx_hash)
+    except KinErrors.CantSimplifyError:
+        raise errors.InvalidTransactionError()
+
+    if tx.operation.type != kin.OperationTypes.PAYMENT:
+        raise errors.InvalidTransactionError()
+
+    # Convert from '2018-11-12T06:45:40Z' to unix timestamp
+    timestamp = datetime.strptime(tx.timestamp, '%Y-%m-%dT%H:%M:%S%z').timestamp()
+    info_response = responses_models.PaymentInfoResponse(tx.source,
+                                                         tx.operation.destination,
+                                                         tx.operation.amount,
+                                                         tx.memo,
+                                                         timestamp)
+
+    return json_response(info_response.to_response_dict(), 200)
+
+
+@app.route('/status', methods=['GET'])
+async def get_status(request):
+    status = await app.kin_account.get_status()
+    status_response = responses_models.StatusResponse(VERSION,
+                                                      status['client']['horizon']['uri'],
+                                                      status['account']['app_id'],
+                                                      status['account']['public_address'],
+                                                      status['account']['balance'],
+                                                      status['account']['channels'])
+
+    return json_response(status_response.to_response_dict(), 200)
+
+
+@app.route('/pay', methods=['POST'])
+@get_model(model=requets_models.PaymentRequest)
+async def pay(payment_request: requets_models.PaymentRequest):
+    try:
+        tx_id = await app.kin_account.send_kin(payment_request.destination,
+                                               payment_request.amount,
+                                               app.minimum_fee,
+                                               payment_request.memo)
+    except KinErrors.AccountNotFoundError:
```

Comment:
will the `internal_error_handler` middleware print also bootstrap errors?
so it will appear in logs as well for debugging?

Reply:
No, errors go to only one error handler

But all of the error handlers print to logs, so it doesn't matter